homeassistant/components/dnsip/config_flow.py

```diff
@@ -0,0 +1,164 @@
+"""Adds config flow for dnsip integration."""
+from __future__ import annotations
+
+import contextlib
+from typing import Any
+
+import aiodns
+from aiodns.error import DNSError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_HOSTNAME,
+    CONF_IPV4,
+    CONF_IPV6,
+    CONF_RESOLVER,
+    CONF_RESOLVER_IPV6,
+    DEFAULT_HOSTNAME,
+    DEFAULT_NAME,
+    DEFAULT_RESOLVER,
+    DEFAULT_RESOLVER_IPV6,
+    DOMAIN,
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOSTNAME, default=DEFAULT_HOSTNAME): cv.string,
+    }
+)
+
+
+async def async_validate_url(
+    url: str, resolver_ipv4: str, resolver_ipv6: str
+) -> dict[str, bool]:
+    """Validate url."""
+    result: dict[str, bool] = {}
+
+    result[CONF_IPV4] = False
+    result[CONF_IPV6] = False
+
+    with contextlib.suppress(DNSError):
+        result[CONF_IPV4] = bool(
+            await aiodns.DNSResolver(nameservers=[resolver_ipv4]).query(url, "A")
+        )
+
+    with contextlib.suppress(DNSError):
+        result[CONF_IPV6] = bool(
+            await aiodns.DNSResolver(nameservers=[resolver_ipv6]).query(url, "AAAA")
+        )
+
```

Comment:
We should do these in parallel with `asyncio.gather` in case the result is a timeout so we don't have to wait for two timeouts